src/Features/Core/Portable/AddParameter/AbstractAddParameterCodeFixProvider.cs

```diff
@@ -113,125 +137,269 @@ internal abstract class AbstractAddParameterCodeFixProvider<
+        {
+            var methodPrefix = methodToUpdate.IsConstructor()
+                ? ""
+                : $"{methodToUpdate.ContainingType.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat)}.";
+            var signature = $"{methodPrefix}{methodToUpdate.Name}({string.Join(", ", parameters)})";
+            var title = string.Format(resourceString, signature);
+            return title;
+        }
 
-        private async Task<Document> FixAsync(
+        private async Task<Solution> FixAsync(
             Document invocationDocument,
             IMethodSymbol method,
             TArgumentSyntax argument,
             SeparatedSyntaxList<TArgumentSyntax> argumentList,
+            bool fixAllReferences,
             CancellationToken cancellationToken)
         {
-            var methodDeclaration = await method.DeclaringSyntaxReferences[0].GetSyntaxAsync(cancellationToken).ConfigureAwait(false);
+            var solution = invocationDocument.Project.Solution;
+            var argumentType = await GetArgumentTypeAsync(invocationDocument, argument, cancellationToken).ConfigureAwait(false);
+            // the argumentNameSuggestion is the base for the parameter name. For each method declaration the name is made unique to avoid name collisions.
+            var (argumentNameSuggestion, isNamedArgument) = await GetNameSuggestionForArgumentAsync(invocationDocument, argument, cancellationToken).ConfigureAwait(false);
+            var referencedSymbols = fixAllReferences
+                ? await FindMethodDeclarationReferences(invocationDocument, method, cancellationToken).ConfigureAwait(false)
+                : method.GetAllMethodSymbolsOfPartialParts();
+            // TODO: Insert hint in the fix with a warning if anySymbolsReferenceNotInSource is true
+            var anySymbolReferencesNotInSource = referencedSymbols.Any(symbol => !symbol.IsFromSource());
+            var locationsInSource = referencedSymbols.Where(symbol => symbol.IsFromSource());
+            // Indexing Locations[0] is valid because IMethodSymbols have one location at most and IsFromSource() tests if there is at least one location.
+            var locationsByDocument = locationsInSource.ToLookup(declarationLocation
+                => solution.GetDocument(declarationLocation.Locations[0].SourceTree));
+            foreach (var documentLookup in locationsByDocument)
+            {
+                var document = documentLookup.Key;
+                var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+                var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                var editor = new SyntaxEditor(syntaxRoot, solution.Workspace);
+                var generator = editor.Generator;
+                foreach (var methodDeclaration in documentLookup)
+                {
+                    var methodNode = syntaxRoot.FindNode(methodDeclaration.Locations[0].SourceSpan);
```

Comment:
just to check.  we have tests that add a parameter to the containing method, right?

Reply:
*containing method*? I don't know what you mean by that.

Reply:
i.e. if i have this:

```c#
void Foo() {
    Foo("bar");
}
```

Reply:
> we have tests that add a parameter to the containing method, right?

Yes, we do have [TestInvocationRecursion](https://github.com/MaStr11/roslyn/blob/306c2b8a0f3fa4b9e66b20b3e2a4121b6bffb890/src/EditorFeatures/CSharpTest/AddParameter/AddParameterTests.cs#L962-L981).